libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/DecodeUtils.kt

```diff
@@ -94,6 +95,53 @@ object DecodeUtils {
         }.orEmpty()
     }
 
+    /**
+     * Decodes geometries of all [RouteLeg]s in a [DirectionsRoute] to [LineString]s
+     * and caches the results.
+     * No decoding is performed for [RouteLeg]s, that already have a cached result available.
+     *
+     * @return decoded [LineString]s. The resulting collection is a nested list of:
+     * ```
+     * [ legs
+     *   { flat steps' geometry decoded to a line string }
+     * ]
+     * ```
+     */
+    @JvmStatic
+    fun DirectionsRoute.legsGeometryToLineString(): List<LineString> {
+        val precision = precision()
+        cacheRoute(route = this, precision)
+        return legs()?.flatMap { leg ->
+            leg.steps()?.map { step ->
+                val points = stepsGeometryDecodeCache.getOrDecode(step.geometry(), precision)
+                LineString.fromLngLats(points)
+            }.orEmpty()
+        }.orEmpty()
+    }
+
+    /**
+     * Decodes geometries of all [RouteLeg]s in a [DirectionsRoute] to [List]s of [List] of [Point]s
+     * and caches the results.
+     * No decoding is performed for [RouteLeg]s, that already have a cached result available.
+     *
+     * @return decoded [List]s of [List] of [Point]s. The resulting collection is a nested list of:
+     * ```
+     * [ legs
+     *   { flat steps' geometry decoded to a list of points }
+     * ]
+     * ```
+     */
+    @JvmStatic
+    fun DirectionsRoute.legsGeometryToPoints(): List<List<Point>> {
+        val precision = precision()
+        cacheRoute(route = this, precision)
+        return legs()?.map { leg ->
+            leg.steps()?.flatMap { step ->
+                stepsGeometryDecodeCache.getOrDecode(step.geometry(), precision)
+            }.orEmpty()
+        }.orEmpty()
+    }
+
```

Comment:
Why do we add this API in the context of this PR? Why is it necessary for SDK's users?

Reply:
It seemed logical to add as we have the same method for step and route. If you think that's redundant, I can make it internal, but why not?

Reply:
I'm asking because I can't find any usages. And I'm wondering, does this code related to this PR? 🤔 
Can you please help me find where it's used or move the functions to a separate PR?

Reply:
Oh, I see. I introduced it but it turned out I don't need it because NN has all the necessary info.